sympy/sets/tests/test_sets.py

```diff
@@ -1597,6 +1600,50 @@ def test_DisjointUnion_len():
     assert len(DisjointUnion(S.EmptySet, S.EmptySet, FiniteSet(x, y, z), S.EmptySet)) == 3
     raises(ValueError, lambda: len(DisjointUnion(Interval(0, 1), S.EmptySet)))
 
+def test_SetKind_ProductSet():
+    p = ProductSet(FiniteSet(Matrix([1, 2])), FiniteSet(Matrix([1, 2])))
+    mk = MatrixKind(NumberKind)
+    k = SetKind(TupleKind(mk, mk))
+    n =  ProductSet(Interval(1, 2), FiniteSet(Matrix([1, 2]))).kind
+    q = SetKind(TupleKind(NumberKind, mk))
+    assert p.kind is k
+    assert n is q
```

Comment:
It's better not to splice tests like this. Assign to `n` and `q` after the first assert so the two cases are clearly separate.